src/file_part.rs

```diff
@@ -0,0 +1,144 @@
+use crate::errors::AppError;
+use crate::json_part::read_page_to_json_str_headlines;
+use log::{error, info};
+use serde_json::{self, Value};
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+use std::io::Write;
+
+const FILE1: &str = "temp_new.json";
+const FILE2: &str = "temp_old.json";
+
+/// Writes headlines to a JSON file.
+///
+/// This function writes the provided headlines to a JSON file. It converts the headlines into
+/// a JSON string using `serde_json::to_string()` and writes the string to the specified file.
+/// Returns `Ok(())` if the operation succeeds, otherwise returns an `io::Error`.
+pub async fn write_headlines_to_json_file(headlines: Vec<String>) -> Result<(), AppError> {
+    info!("Writing headlines to JSON file.");
+    let json_str = serde_json::to_string(&headlines).map_err(AppError::ParseJsonError)?;
+
+    let mut file = File::create("temp_new.json")?;
```

Comment:
Can this be replaced?
```suggestion
    let mut file = File::create(FILE1)?;
```